### PR TITLE
Fixes #6867 - describe module search behavior

### LINKED ISSUE #6867: How $env:PSModulePath is used to find modules should be explained
I thought that if I added a path like "C:\src" to $env:PSModulePath, then it would make modules such as the following automatically discoverable/loadable:

    C:\src\foo\foo.psd1
    C:\src\foo\1.2.3\foo.psd1

And indeed, that works. But what I did NOT know is that PSModulePath apparently does a complete subtree search of any dir you put into it. For example, this module would ALSO become available:

    C:\src\fee\fie\foe\fum\fum.psm1

I did not expect that.

We should document how PowerShell uses the value of $env:PSModulePath to find modules.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9e779ca4-dbd0-480a-ccb8-ee65df2da1c0
* Version Independent ID: 0e9c150c-ad17-86a9-f0e1-68c90cd93ead
* Content: [about_PSModulePath - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_psmodulepath?view=powershell-7)
* Content Source: [reference/7.0/Microsoft.PowerShell.Core/About/about_PSModulePath.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.0/Microsoft.PowerShell.Core/About/about_PSModulePath.md)
* Product: **powershell**
* Technology: **powershell-conceptual**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1792028](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1792028) - Fixes #6867 - describe module search behavior

## PR Context

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Version 7.1 preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
 online version: https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_PSModulePath?view=powershell-5.1&WT.mc_id=ps-gethelp
 schema: 2.0.0
 title: about_PSModulePath
 ---
 # About PSModulePath
 
 The `$env:PSModulePath` environment variable contains a list of folder
-locations that are searched to find modules and resources.
+locations that are searched to find modules and resources. PowerShell
+recursively search each folder for module (`.psd1` or `.psm1`) files.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md:14]: searches

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md [hunk 2]
```diff
@@ -179,6 +180,26 @@ The PowerShell 7 startup continues as-is with the addition of inheriting paths
 that PowerShell Core 6 added. Since the PS7-specific paths are prefixed, there
 is no functional issue.
 
+## Module search behavior
+
+PowerShell recursively searches each folder in the PSModulePath for module
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md:185]: Format PSModulePath

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md [hunk 3]
```diff
@@ -179,6 +180,26 @@ The PowerShell 7 startup continues as-is with the addition of inheriting paths
...
+(`.psd1` or `.psm1`) files. This allows multiple versions of the same module to
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md:186]: Give `this` a subject noun for loc.

**Review by DCtheGeek — CHANGES_REQUESTED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md:14]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_PSModulePath.md:185]: fixed

**Review by DCtheGeek — APPROVED** [review]

**Pull request merged**